packages/gatsby-recipes/src/providers/npm/package-json.js

```diff
@@ -5,6 +5,63 @@ const getDiff = require(`../utils/get-diff`)
 
 const resourceSchema = require(`../resource-schema`)
 
+class Deferred {
```

Comment:
This class is cool and seems useful.  Is it or could it be used elsewhere?

Reply:
possibly! I copied it off stackoverflow haha — we'll probably migrate this whole bit of code that handles writing to a single shared resource (in this case to package.json) to its own library.